jumble/src/components/Slot.tsx

```diff
@@ -0,0 +1,105 @@
+// @ts-nocheck vendored
+/**
+Vendored from https://gist.github.com/giuseppelt/14d2bda071f728f5164baa76ecf60b01
+
+Container component:
+```
+import { Slot } from "./components";
+
+export function TestContainer({ children } ) {
+  return (
+    <div>
+      <Slot name="header" required children={children} />
+      <p>content</p>
+      <Slot children={children} />
+      <Slot name="footer" fallback={<p>Default Footer</p>} children={children} />
+    </div>
+  );
+}
+```
+
+Using component:
+```
+import { TestContainer } from "./components";
+
+export function ComplexComponent() {
+   return (
+      <TextContainer>
+        <p slot="header">This will be the header</p>
+
+        <p>Other body content</p>
+        <p>Other body content</p>
+        <p>Other body content</p>
+      </TextContainer>
+   );
+}
```

Comment:
Cool! This is a nice alternative to the regular React pattern I've seen (passing fragments around as props)